pipeline/git_preflight.go

```
@@ -0,0 +1,324 @@
+		return fmt.Errorf("%w: workdir is filesystem root (%s).\n\n  cd into a project subdirectory first.", ErrGitAutoInitRefused, abs)
+	}
+	// Use safetyLatches's `--git-dir` check (NOT `--is-inside-work-tree` like
+	// the requires:git satisfaction probe in checkGit) because we want to
+	// catch bare repos, linked worktrees, and submodules — all three resolve
+	// a git dir even though work-tree-only operations would later fail.
+	// Different questions: checkGit asks "would `git commit` work here?"
+	// (work-tree required), safetyLatches asks "is this any kind of nested
+	// git context where running `git init` would create a confusing
+	// duplicate?" (any git-dir resolution counts).
+	if _, lerr := exec.LookPath("git"); lerr != nil {
+		return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 274) but it returns nil

(nilerr)